dev/kyuubi-extension-spark-3-1/src/test/scala/org/apache/spark/sql/KyuubiExtensionSuite.scala

```diff
@@ -1303,4 +1303,96 @@ class KyuubiExtensionSuite extends KyuubiSparkSQLExtensionTest {
       }
     }
   }
+
+  test("test watchdog with query forceMaxOutputRows") {
+
+    withSQLConf(KyuubiSQLConf.WATCHDOG_FORCED_MAXOUTPUTROWS.key -> "10") {
+
+      assert(sql("SELECT * FROM VALUES(1, 'a'),(2, 'b') AS t(c1, c2)")
+        .queryExecution.analyzed.isInstanceOf[GlobalLimit])
+
+      assert(sql("SELECT * FROM VALUES(1, 'a'),(2, 'b') AS t(c1, c2) LIMIT 1")
+        .queryExecution.analyzed.asInstanceOf[GlobalLimit].maxRows.contains(1))
+
+      assert(sql("SELECT * FROM VALUES(1, 'a'),(2, 'b') AS t(c1, c2) LIMIT 11")
+        .queryExecution.analyzed.asInstanceOf[GlobalLimit].maxRows.contains(10))
+
+      assert(!sql("SELECT count(*) FROM VALUES(1, 'a'),(2, 'b') AS t(c1, c2)")
+        .queryExecution.analyzed.isInstanceOf[GlobalLimit])
+
+      assert(sql(
+        """
+          |SELECT c1, COUNT(*)
+          |FROM VALUES(1, 'a'), (2, 'b') AS t(c1, c2)
+          |GROUP BY c1
+          |""".stripMargin).queryExecution.analyzed.isInstanceOf[GlobalLimit])
```

Comment:
It's flaky with Spark3.2.0 that the issue of some logical plan has not overridden maxRows is fixed. see https://github.com/apache/spark/pull/30443.

We can use the test table, see `KyuubiExtensionSuite.setupData`.

Reply:
> It's flaky with Spark3.2.0 that the issue of some logical plan has not overridden maxRows is fixed. see [apache/spark#30443](https://github.com/apache/spark/pull/30443).
> 
> We can use the test table, see `KyuubiExtensionSuite.setupData`.

_见贤思齐_
I read the [apache/spark#30443](https://github.com/apache/spark/pull/30443). It seems to get more exactly maxRows in Optimizer? I don't understand how affect the test case changed, as `VALUES(1, 'a'), (2, 'b') AS t(c1, c2)`

Reply:
I have replaced it with setupData

Reply:
> > It's flaky with Spark3.2.0 that the issue of some logical plan has not overridden maxRows is fixed. see [apache/spark#30443](https://github.com/apache/spark/pull/30443).
> > We can use the test table, see `KyuubiExtensionSuite.setupData`.
> 
> _见贤思齐_
> I read the [apache/spark#30443](https://github.com/apache/spark/pull/30443). It seems to get more exactly maxRows in Optimizer? I don't understand how affect the test case changed yet, as `VALUES(1, 'a'), (2, 'b') AS t(c1, c2)`?

Reply:
This query used the plan `LocalRelation` and after 3.2.0 the `LocalRelation`'s maxRows will be the data size. That said, if you have a query `SELECT * FROM VALUES(1, 'a'), (2, 'b') AS t(c1, c2)`, the root Project node maxRows will be `Some(2)` instead of `None`. Then the test will fail if we switch the Spark version from 3.1 to 3.2.

Reply:
> This query used the plan `LocalRelation` and after 3.2.0 the `LocalRelation`'s maxRows will be the data size. That said, if you have a query `SELECT * FROM VALUES(1, 'a'), (2, 'b') AS t(c1, c2)`, the root Project node maxRows will be `Some(2)` instead of `None`. Then the test will fail if we switch the Spark version from 3.1 to 3.2.
Clear! Thanks for your advice!